assets/questions/reverseWords.js

```diff
@@ -28,124 +28,388 @@ globalData.questions['reverseWords'] = {  // eslint-disable-line dot-notation
+      buggyFunctionName: 'AuxiliaryCode.elideNonLetters',
+      messages: [
+        [
+          'Try running your algorithm, by hand, on the sample input. Does it ',
+          'give the right answer?'
+        ].join(''),
+        [
+          "Please check that you're handling spaces correctly."
+        ].join(''),
+        [
+          'Spaces and punctuation need to be preserved. If there is a space ',
+          'between two words in the input, there should also be a space ',
+          'between those words in the output.'
+        ].join('')
+      ]
     }, {
-      input: '123 456 789 ',
-      allowedOutputs: ['321 654 987 '],
-      tag: 'numbers'
+      buggyFunctionName: 'AuxiliaryCode.forgetLastWord',
+      messages: [
+        [
+          'Try running your algorithm, by hand, on the sample input. Does it ',
+          'give the right answer?'
+        ].join(''),
+        [
+          "It looks like you're returning too few words. Did you forget to ",
+          "add the last one?"
+        ].join('')
+      ]
     }, {
-      input: 'test for dashes-and others',
-      allowedOutputs: ['tset rof dna-sehsad srehto'],
-      tag: 'dashed words'
+      buggyFunctionName: 'AuxiliaryCode.forgetToReverseLastWord',
+      messages: [
+        [
+          'Try running your algorithm, by hand, on the sample input. Does it ',
+          'give the right answer?'
+        ].join(''),
+        [
+          'When your code is run with "Hello, John", where does the last ',
+          'character end up?'
+        ].join(''),
+        'Looks like you may have forgotten to reverse the last word...'
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.treatCommaAsPartOfWord',
+      messages: [
+        [
+          'When you run your algorithm with "Hello, John", what is the first ',
+          'character in the output? Is it correct?'
+        ].join(''),
+        [
+          'Remember, per the question, that commas, spaces, etc. should be ',
+          'left in place.'
+        ].join(''),
+        [
+          'Your code seems to treat "Hello," as one word, but the comma at ',
+          'the end is not part of the word.'
+        ].join('')
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.addSpaceAtEnd',
+      messages: [
+        [
+          'Adding spaces at ends of words to keep things simple is a nice ',
+          'idea, but be careful about how you handle the extra space at the ',
+          'end.'
+        ].join('')
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.treatAsciiCharsContiguously',
+      messages: [
+        [
+          'Check the ASCII tables -- are you sure that "A"-"z" be treated as ',
+          'a contiguous block of characters?'
+        ].join('')
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.useWeakInequalitiesForCharValidity',
+      messages: [
+        [
+          'Are you correctly handling the check for whether a character is a ',
```

Comment:
Does there need to be an "or not" in here? I can't tell, it just feels like it's missing for me

Reply:
I checked -- consensus [here](https://english.stackexchange.com/questions/3382/whether-or-not-vs-whether) seems to be to just use "whether", so I'm going to go for that.